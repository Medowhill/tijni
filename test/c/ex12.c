int f(int **x) {
  return **x;
}

int g(int **x, int **y, int z) {
  int **p = x;
  if (z) p = y;
  return f(p);
}
