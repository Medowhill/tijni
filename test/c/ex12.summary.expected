{
  "f": {
    "params": [ { "name": "%x(f)", "used": true, "freed": false } ],
    "ret": [ "'''%x(f)" ],
    "env": {
      "%x(f)": [ "'%x(f)" ],
      "%0(f)": [ "''%x(f)" ],
      "%1(f)": [ "'''%x(f)" ],
      "'%x(f)": [ "''%x(f)" ],
      "''%x(f)": [ "'''%x(f)" ]
    }
  },
  "g": {
    "params": [ { "name": "%x(g)", "used": true, "freed": false }, { "name": "%y(g)", "used": true, "freed": false }, { "name": "%z(g)", "used": false, "freed": false } ],
    "ret": [ "'''%x(g)", "'''%y(g)" ],
    "env": {
      "%x(f)": [ "'%x(g)", "'%y(g)" ],
      "%0(f)": [ "''%x(g)", "''%y(g)" ],
      "%1(f)": [ "'''%x(g)", "'''%y(g)" ],
      "%x(g)": [ "'%x(g)" ],
      "%y(g)": [ "'%y(g)" ],
      "%z(g)": [ "'%z(g)" ],
      "%tobool(g)": [],
      "%p.0(g)": [ "'%x(g)", "'%y(g)" ],
      "%call(g)": [ "'''%x(g)", "'''%y(g)" ],
      "'%x(g)": [ "''%x(g)", "''%y(g)" ],
      "'%y(g)": [ "''%x(g)", "''%y(g)" ],
      "''%x(f)": [ "'''%x(g)", "'''%y(g)" ]
    }
  }
}